### PR TITLE
Add some useful http's response procs

### PR DESCRIPTION
**Pull request by data-man**

### CODE REVIEW

File: lib/pure/httpclient.nim [hunk 1]
```diff
@@ -150,6 +150,33 @@ proc code*(response: Response | AsyncResponse): HttpCode
...
+
+proc contentLength*(response: Response | AsyncResponse): int =
+  ## Retrieves the specified response's ``Content length``.
+  ## If the response's ``content length`` does not have a
+  ## corresponding value then -1 will be returned.
+  ##
+  var contentLengthHeader = response.headers.getOrDefault("Content-Length")
+  try:
+    result = contentLengthHeader.parseint()
+  except:
```
Comment (Araq) [inline lib/pure/httpclient.nim]: ``except ValueError``  but better use parseutils.parseSaturatedNatural

File: lib/pure/httpclient.nim [hunk 2]
```diff
@@ -150,6 +150,33 @@ proc code*(response: Response | AsyncResponse): HttpCode
...
+
+proc lastModified*(response: Response | AsyncResponse): DateTime =
+  ## Retrieves the specified response's ``Last modified time``.
+  ## If the response's ``Last modified time`` does not have a
+  ## corresponding value then "01 Jan 0000 00:00:00" will be returned.
+  ##
+  var lastModifiedHeader = response.headers.getOrDefault("last-modified")
+  try:
+    result = parse(lastModifiedHeader, "ddd, dd MMM yyyy HH:mm:ss Z")
+  except:
```
Comment (Araq) [inline lib/pure/httpclient.nim]: ``except ValueError``

Comment (data-man) [reply lib/pure/httpclient.nim]: Raise exception instead of returning a special value?

File: lib/pure/httpclient.nim [hunk 3]
```diff
@@ -150,6 +150,33 @@ proc code*(response: Response | AsyncResponse): HttpCode
   ## corresponding ``HttpCode``.
   return response.status[0 .. 2].parseInt.HttpCode
 
+proc contentType*(response: Response | AsyncResponse): string =
+  ## Retrieves the specified response's ``Content type``.
```
Comment (dom96) [inline lib/pure/httpclient.nim]: Nitpick: don't use \`\` in documentation where quotes would be more appropriate.

Comment (dom96) [reply lib/pure/httpclient.nim]:
```
## Retrieves the specified response's content type. This is effectively the value of the "content-type" header.
```

You don't even need to emphasise the "content type" really.

**Comment by dom96** [comment]
@Araq will probably disagree with me but I think these should raise exceptions.

File: lib/pure/httpclient.nim [hunk 4]
```diff
@@ -150,6 +150,31 @@ proc code*(response: Response | AsyncResponse): HttpCode
...
+  ## This is effectively the value of the "Content-Length" header.
+  ##
+  var contentLengthHeader = response.headers.getOrDefault("Content-Length")
+  discard contentLengthHeader.parseSaturatedNatural(result)
+
+proc lastModified*(response: Response | AsyncResponse): DateTime =
+  ## Retrieves the specified response's last modified time.
+  ## This is effectively the value of the "last-modified" header.
+  ## If the response's does not have a corresponding value
+  ## then "01 Jan 0000 00:00:00" will be returned.
```
Comment (Araq) [inline lib/pure/httpclient.nim]: Meh, nowhere else do we use a "null" DateTime. This should just raise the ValueError that ``parse`` raises.

Comment (data-man) [reply lib/pure/httpclient.nim]: Done.
Also I fixed date's parsing bug.
> Error: unhandled exception: index out of bounds [IndexError]

if token's value is empty.

**Comment by dom96** [comment]
Argh, please don't fix bugs in something completely different in the same PR.

**Comment by dom96** [comment]
I merged this manually, please make a separate PR for the times fix (with a test case)

**Pull request closed without merging**